R/oauth-flow-auth-code.R

```diff
@@ -233,6 +227,61 @@ oauth_flow_auth_code <- function(client,
   )
 }
 
+normalize_redirect_uri <- function(redirect_uri,
+                                   host_name = deprecated(),
+                                   host_ip = deprecated(),
+                                   port = deprecated(),
+                                   error_call = caller_env()) {
+
+  parsed <- url_parse(redirect_uri)
+
+  if (lifecycle::is_present(host_name)) {
+    lifecycle::deprecate_warn(
+      when = "0.3.0",
+      what = "oauth_flow_auth_code(host_name)",
+      with = "oauth_flow_auth_code(redirect_uri)"
+    )
+    parsed$hostname <- host_name
+  }
+
+  if (lifecycle::is_present(port)) {
+    lifecycle::deprecate_warn(
+      when = "0.3.0",
+      what = "oauth_flow_auth_code(port)",
+      with = "oauth_flow_auth_code(redirect_uri)"
+    )
+    parsed$port <- port
+  }
+
+  if (lifecycle::is_present(host_ip)) {
+    lifecycle::deprecate_warn("0.3.0", "oauth_flow_auth_code(host_ip)")
+  }
+
+  localhost <- parsed$hostname == "localhost"
```

Comment:
Feels like you might one day have to ponder what happens if someone provides `urn:ietf:wg:oauth:2.0:oob` as `redirect_uri`.

Reply:
Yeah and that seems like it will be a relatively straighforward change to make.